receiver/stanzareceiver/converter.go

```diff
@@ -0,0 +1,220 @@
+	case uint32:
+		dest.SetIntVal(int64(t))
+	case uint16:
+		dest.SetIntVal(int64(t))
+	case uint8:
+		dest.SetIntVal(int64(t))
+	case uint:
+		dest.SetIntVal(int64(t))
+	case float64:
+		dest.SetDoubleVal(t)
+	case float32:
+		dest.SetDoubleVal(float64(t))
+	case map[string]interface{}:
+		dest.SetMapVal(toAttributeMap(t))
+	case []interface{}:
+		dest.SetMapVal(toAttributeMap(sliceToMap(t)))
+	default:
+		dest.SetStringVal(fmt.Sprintf("%v", t))
+	}
+}
+
+func toAttributeMap(obsMap map[string]interface{}) pdata.AttributeMap {
+	attMap := pdata.NewAttributeMap()
+	attMap.InitEmptyWithCapacity(len(obsMap))
+	for k, v := range obsMap {
+		switch t := v.(type) {
+		case bool:
+			attMap.InsertBool(k, t)
+		case string:
+			attMap.InsertString(k, t)
+		case []byte:
+			attMap.InsertString(k, string(t))
+		case int64:
+			attMap.InsertInt(k, t)
+		case int32:
+			attMap.InsertInt(k, int64(t))
+		case int16:
+			attMap.InsertInt(k, int64(t))
+		case int8:
+			attMap.InsertInt(k, int64(t))
+		case int:
+			attMap.InsertInt(k, int64(t))
+		case uint64:
+			attMap.InsertInt(k, int64(t))
+		case uint32:
+			attMap.InsertInt(k, int64(t))
+		case uint16:
+			attMap.InsertInt(k, int64(t))
+		case uint8:
+			attMap.InsertInt(k, int64(t))
+		case uint:
+			attMap.InsertInt(k, int64(t))
+		case float64:
+			attMap.InsertDouble(k, t)
+		case float32:
+			attMap.InsertDouble(k, float64(t))
+		case map[string]interface{}:
+			subMap := toAttributeMap(t)
+			subMapVal := pdata.NewAttributeValueMap()
+			subMapVal.SetMapVal(subMap)
+			attMap.Insert(k, subMapVal)
+		case []interface{}:
+			subMap := toAttributeMap(sliceToMap(t))
+			subMapVal := pdata.NewAttributeValueMap()
+			subMapVal.SetMapVal(subMap)
+			attMap.Insert(k, subMapVal)
+		default:
+			attMap.InsertString(k, fmt.Sprintf("%v", t))
+		}
+	}
+	return attMap
+}
+
+// This returns a map of stringified index to value,
+// rather than an array because the pdata package does not support arrays
+func sliceToMap(arr []interface{}) map[string]interface{} {
+	sliceAsMap := make(map[string]interface{})
+	for i, v := range arr {
+		sliceAsMap[strconv.Itoa(i)] = v
+	}
+	return sliceAsMap
+}
+
+func convertSeverity(s entry.Severity) (string, pdata.SeverityNumber) {
+	switch {
+
+	// Handle standard severity levels
+	case s == entry.Catastrophe:
+		return "Fatal", pdata.SeverityNumberFATAL4
+	case s == entry.Emergency:
+		return "Error", pdata.SeverityNumberFATAL
+	case s == entry.Alert:
+		return "Error", pdata.SeverityNumberERROR3
+	case s == entry.Critical:
+		return "Error", pdata.SeverityNumberERROR2
+	case s == entry.Error:
+		return "Error", pdata.SeverityNumberERROR
+	case s == entry.Warning:
+		return "Info", pdata.SeverityNumberINFO4
+	case s == entry.Notice:
+		return "Info", pdata.SeverityNumberINFO3
+	case s == entry.Info:
+		return "Info", pdata.SeverityNumberINFO
+	case s == entry.Debug:
+		return "Debug", pdata.SeverityNumberDEBUG
+	case s == entry.Trace:
+		return "Trace", pdata.SeverityNumberTRACE2
```

Comment:
Are the pdata levels chosen arbitrary or is there some kind of logic to this?  That's fine if it is arbitrary I'm just curious.

Reply:
Ok, I think I kind of see how the official levels are interspersed with the custom ones.

Reply:
I wouldn't say the conversion is arbitrary, but there's definitely some subjectivity to it.

Just trying to map from a 0-100 representation to a 0-24 representation, and it gets a little crowded around some levels.